### PR TITLE
feat(styles): implement dark theme and updated design for options menu dropdown

### PR DESCRIPTION
**Pull request by scurker**

### CODE REVIEW

File: packages/styles/dropdown.css [hunk 1]
```diff
@@ -1,44 +1,3 @@
...
-  box-shadow: 0px 0px 2px 1px rgba(0, 0, 0, 0.3);
-  border: 1px solid var(--field-border-color);
-  color: var(--text-color-base);
-  display: none;
-  list-style: none;
-}
-
-.Dropdown.Dropdown--active {
-  display: block;
-}
```
Comment (thuey) [inline packages/styles/dropdown.css:14]: Should this be removed, too? https://github.com/dequelabs/cauldron/blob/7fd7d095bc8c85c5369371baaf74c7300c578fd2/packages/react/src/components/TopBar/TopBarMenu.tsx#L104

File: packages/styles/dropdown.css [hunk 2]
```diff
@@ -1,44 +1,3 @@
-.Dropdown {
```
Comment (thuey) [inline packages/styles/dropdown.css:1]: As far as I can tell, these styles weren't documented anywhere, but is there any concern about this being a breaking change if any code was using `.Dropdown`?

**Review by thuey — DISMISSED** [review]

File: packages/styles/dropdown.css (hunk 1 above)
Comment (scurker) [reply packages/styles/dropdown.css:14]: No, this prevents a breaking change for someone using that class.

File: packages/styles/dropdown.css (hunk 2 above)
Comment (scurker) [reply packages/styles/dropdown.css:1]: I'm not 100% sure why those styles were even there. They probably should've exclusively been a part of the OptionsMenu styles, but I'm leaving them there for now.

**Review by thuey — APPROVED** [review]

**Pull request merged**